nanoFramework.Graphics.Core/System/Drawing/Rectangle.cs

```diff
@@ -0,0 +1,354 @@
+        /// <see cref='Rectangle'/> .
+        /// </summary>
+        public readonly bool Contains(int x, int y) => X <= x && x < X + Width && Y <= y && y < Y + Height;
+
+        /// <summary>
+        /// Determines if the specified point is contained within the rectangular region defined by this
+        /// <see cref='Rectangle'/> .
+        /// </summary>
+        public readonly bool Contains(Point pt) => Contains(pt.X, pt.Y);
+
+        /// <summary>
+        /// Determines if the rectangular region represented by <paramref name="rect"/> is entirely contained within the
+        /// rectangular region represented by this <see cref='Rectangle'/> .
+        /// </summary>
+        public readonly bool Contains(Rectangle rect) =>
+            (X <= rect.X) && (rect.X + rect.Width <= X + Width) &&
+            (Y <= rect.Y) && (rect.Y + rect.Height <= Y + Height);
+
+        /// <inheritdoc/>
+        public readonly override int GetHashCode()
+        {
+            // Original: HashCode.Combine(X, Y, Width, Height);
+            unchecked
+            {
+                return X.GetHashCode() ^ Y.GetHashCode() ^ Width.GetHashCode() ^ Height.GetHashCode();
+            }
+        }
+
+        /// <summary>
+        /// Inflates this <see cref='Rectangle'/> by the specified amount.
+        /// </summary>
+        public void Inflate(int width, int height)
+        {
+            unchecked
+            {
+                X -= width;
+                Y -= height;
+
+                Width += 2 * width;
+                Height += 2 * height;
+            }
+        }
+
+        /// <summary>
+        /// Inflates this <see cref='Rectangle'/> by the specified amount.
+        /// </summary>
+        public void Inflate(Size size) => Inflate(size.Width, size.Height);
+
+        /// <summary>
+        /// Creates a <see cref='Rectangle'/> that is inflated by the specified amount.
+        /// </summary>
+        public static Rectangle Inflate(Rectangle rect, int x, int y)
+        {
+            var r = rect;
+            r.Inflate(x, y);
+            return r;
+        }
+
+        /// <summary>
+        /// Creates a Rectangle that represents the intersection between this Rectangle and rect.
+        /// </summary>
+        public void Intersect(Rectangle rect)
+        {
+            var result = Intersect(rect, this);
+
+            X = result.X;
+            Y = result.Y;
+            Width = result.Width;
+            Height = result.Height;
+        }
+
+        /// <summary>
+        /// Creates a rectangle that represents the intersection between a and b. If there is no intersection, an
+        /// empty rectangle is returned.
+        /// </summary>
+        public static Rectangle Intersect(Rectangle a, Rectangle b)
+        {
+            // TODO: The documentation for Graphics says the Math library is required but it's not referenced at this time
+            int x1 = Mathematics.Max(a.X, b.X);// Math.Max(a.X, b.X);
```

Comment:
Min and Max are available in mscorlib as `private` for situations like this one.
Just need to add nanoFramework.Graphics to "Friends" list.